lbr_demos/lbr_demos_fri_ros2_cpp/src/cartesian_path_planning_node.cpp

```diff
@@ -0,0 +1,217 @@
+
+class JointPositionPublisher:public rclcpp::Node
+{
+	private:
+		rclcpp::Publisher<lbr_fri_msgs::msg::LBRPositionCommand>::SharedPtr publisher_;
+        rclcpp::Subscription<lbr_fri_msgs::msg::LBRState>::SharedPtr subscriber_;
+
+        KDL::Chain chain_; // robot kinematics chain exetracted from robot URDF file
+
+        double amplitude_; // rad
+        double frequency_; // Hz
+        double sampling_time_; // sampling time for the state of robot and sending position command
+        double phase_; // initial phase
+		int count_;
+
+	private:
+        /**
+         * @function: callback function for Robot State Subscriber
+        */
+        void topic_callback(const lbr_fri_msgs::msg::LBRState& msg)
+        {
+            current_robot_state = msg;
+            
+            if(count_ < 200) // in first 2 seconds, to make 'initial_position_command' be 'measured_joint_position'
+            {
+                std::cout << "count_" << count_ << std::endl;
+                double joint_position[7];
+                for(int i = 0; i < 7; i++)
+                {
+                    joint_position[i] = current_robot_state.measured_joint_position[i]; 
+                }
+                initial_cartesian_position = computeForwardKinematics(chain_, joint_position);
+            }
+            else
+            {
+                if (current_robot_state.session_state == KUKA::FRI::COMMANDING_ACTIVE) 
+                {
+                    KDL::Frame cartesian_position_command = initial_cartesian_position;
+
+                    cartesian_position_command.p = KDL::Vector(initial_cartesian_position.p.x(), initial_cartesian_position.p.y(), initial_cartesian_position.p.z() + amplitude_ * sin(phase_));
+
+                    phase_ = phase_ + 2 * M_PI * frequency_ * sampling_time_;
+
+                    unsigned int joint_number = chain_.getNrOfJoints(); // for kuka iiwa7 robot, this will be 7 
+                    KDL::JntArray current_joint_positions(joint_number);
+
+                    for(unsigned int i = 0; i < joint_number; i++)
+                    {
+                        current_joint_positions(i) = current_robot_state.measured_joint_position[i];
+                    }
+
+                    lbr_fri_msgs::msg::LBRPositionCommand joint_position_command = computeInverseKinematics(chain_, cartesian_position_command, current_joint_positions);
+
+                    publisher_->publish(joint_position_command);
+                }
+            }
+
+            count_++;
+
+            return;
+        }
+
+	public:
+		JointPositionPublisher():Node("joint_position_publisher_node"), amplitude_(0.05), frequency_(0.5), sampling_time_(0.01)
+		{
+            phase_ = 0.0; 
+            count_= 0.0;
+
+            std::string urdf_file_path = "/home/nearlab-iiwa/lbr-stack/src/lbr_fri_ros2_stack/lbr_description/urdf/iiwa7/my_iiwa7.urdf"; // path of your robot urdf file
```

Comment:
would it be possible to load this from the `/robot_description` parameter instead